config/routes.rb

```diff
@@ -1,27 +1,29 @@
 Rails.application.routes.draw do
+  # Unauthenticated routes (Generators place them here. Watch out!)
 
-  resources :assignments
   resources :volunteer_applications, only: [:new, :create] do
     get :thanks, on: :collection
   end
 
-  # Authenticated routes start here
+  # Authenticated routes start here (Watch out!)
   devise_for :users
```

Comment:
Hmm the `devise_for` just adds the routes from Devise, it doesn't automagically add authentication to following routes. Or maybe I misunderstand something?

Reply:
Thats how I've interpreded it the other day. But I didn't dig deeper. Hope I didn't produce fake news here ;)

Reply:
It definitely makes sense to group authorized/unauthorized routes here, but the actual authorization needs to be performed in each controller so the "Watch out!" maybe sends the wrong signal, and we already have `:verify_authorized` watching out for us :)

Reply:
I did produce fake news and now where both mixing up things 😉 
`:verify_authorized` is Pundits method, and it doesn't bother about logged in or not (exept, that it requires its @user class variable of course)

My fake news is, that the authentication (login) is triggered by `before_action :authenticate_user!` and in the controllers where authentication is not required it is prevented from beeing triggered by `skip_before_action :authenticate_user!`

So there is no harm or benefit to place these things anywhere in the routes what so ever ... 😉

Reply:
But, yes, having the routes in a order that makes sense is definitely an advise I completely want to follow!